Meta/analyze-qemu-coverage.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+SCRIPT_DIR="$(dirname "${0}")"
+
+if [ -z "$SERENITY_ARCH" ]; then
+    SERENITY_ARCH="i686"
+fi
+
+toolchain_suffix=
+if [ "$SERENITY_TOOLCHAIN" = "Clang" ]; then
+    toolchain_suffix="clang"
+fi
+
+BUILD_DIR="${SCRIPT_DIR}/../Build/${SERENITY_ARCH}${toolchain_suffix}"
+TEMP_PROFDATA="$BUILD_DIR/tmp_profile_data"
+
+mkdir -p "$TEMP_PROFDATA"
+mkdir -p "$BUILD_DIR/mnt"
+
+[ -z "${ELEVATE+x}" ] && ELEVATE="sudo"
+echo "ELEVATE is $ELEVATE"
+
+echo "[ELEVATED] Mounting _disk_image into $BUILD_DIR/mnt, this requires elevated privileges..."
+"$ELEVATE" mount "$BUILD_DIR/_disk_image" "$BUILD_DIR/mnt"
+echo "[ELEVATED] Copying profile data files out of $BUILD_DIR/mnt, this requires elevated privileges..."
+"$ELEVATE" cp -r "$BUILD_DIR/mnt/home/anon/profiles" "$TEMP_PROFDATA/"
+echo "[ELEVATED] Unmounting _disk_image, this requires elevated privileges..."
+"$ELEVATE" umount "$BUILD_DIR/mnt"
+echo "[ELEVATED] Making sure profile data files are owned by the current user, this requires elevated privileges..."
+"$ELEVATE" chown -R "$(id -u)":"$(id -g)" "$TEMP_PROFDATA/"
+
+
+echo "Moving profile data into $TEMP_PROFDATA directly..."
+mv "$TEMP_PROFDATA"/profiles/* "$TEMP_PROFDATA/"
+
+echo "Discovering all binaries and shared libraries in $BUILD_DIR/Root"
+# shellcheck disable=SC2156 # The recommended fix on the Shellcheck github page for this warning causes the script to not find any files at all
+mapfile -d '\n' all_binaries < <(find "$BUILD_DIR"/Root -type f -exec sh -c "file {} | grep -vi relocatable | grep -Eiq ': elf (32|64)-bit'" \; -print | grep -Ev '(usr\/Tests|usr\/local|boot\/|Loader.so)')
+
+CLANG_BINDIR="${SCRIPT_DIR}/../Toolchain/Local/clang/bin"
+
+# FIXME: Come up with our own coverage prep script instead of using llvm's
+if [ ! -f "$BUILD_DIR/prepare-code-coverage-artifact.py" ]; then
+    LLVM_14_RELEASE_HASH=329fda39c507e8740978d10458451dcdb21563be
+    wget https://github.com/llvm/llvm-project/blob/${LLVM_14_RELEASE_HASH}/llvm/utils/prepare-code-coverage-artifact.py -P "$BUILD_DIR"
```

Comment:
Shouldn't we, for security, hash the python script and check that hash?

Reply:
For security? Not sure what security we would gain from hashing the file contents. For integrity maybe, but we already are grabbing the file from a known fixed hash instead of the release tag.

Reply:
Still, someone can MITM the script, as any download. Just that the URL is complex doesn't help IMHO.

Reply:
Welp, glad I went to add this anyway, my download link was wrong, I needed the raw.githubusercontent link instead. We now check its sha256